recipes/bison/all/conanfile.py

```diff
@@ -0,0 +1,164 @@
+    argparse = ArgumentParser()
+    argparse.add_argument("--output", "-o", help="output")
+    known, unknown = argparse.parse_known_args()
+    if known.output:
+        try:
+            open(known.output, "w").write("\\n")
+        except OSError:
+            pass
+"""
+
+
+class BisonConan(ConanFile):
+    name = "bison"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://www.gnu.org/software/bison/"
+    description = "Bison is a general-purpose parser generator"
+    topics = ("conan", "bison", "parser")
+    license = "GPL-3.0-or-later"
+    exports_sources = "patches/**"
+
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "fPIC": [True, False],
+    }
+    default_options = {
+        "fPIC": True,
+    }
+
+    _autotools = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = "bison-" + self.version
+        os.rename(extracted_dir, self._source_subfolder)
+
+    def requirements(self):
+        self.requires("m4/1.4.18")
+
+    def build_requirements(self):
+        if tools.os_info.is_windows and not tools.get_env("CONAN_BASH_PATH") and \
+                tools.os_info.detect_windows_subsystem() != "msys2":
+            self.build_requires("msys2/20190524")
+        if self.settings.compiler == "Visual Studio":
+            self.build_requires("automake/1.16.1")
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+
+    @contextmanager
+    def _build_context(self):
+        if self.settings.compiler == "Visual Studio":
+            with tools.vcvars(self.settings):
+                build_env = {
+                    "CC": "{} cl -nologo".format(tools.unix_path(self.deps_user_info["automake"].compile)),
+                    "CXX": "{} cl -nologo".format(tools.unix_path(self.deps_user_info["automake"].compile)),
+                    "CFLAGS": "-{}".format(self.settings.compiler.runtime),
+                    "LD": "link",
+                    "NM": "dumpbin -symbols",
+                    "STRIP": ":",
+                    "AR": "{} lib".format(tools.unix_path(self.deps_user_info["automake"].ar_lib)),
+                    "RANLIB": ":",
+                }
+                with tools.environment_append(build_env):
+                    yield
+        else:
+            yield
+
+    def _configure_autotools(self):
+        if self._autotools:
+            return self._autotools
+        self._autotools = AutoToolsBuildEnvironment(self, win_bash=tools.os_info.is_windows)
+        args = [
+            "HELP2MAN={}".format(tools.unix_path(self._simple_output_script)),
+            "MAKEINFO={}".format(tools.unix_path(self._simple_output_script)),
```

Comment:
If I understood this well, this is done to "replace" help2man and makeinfo tools, right?

Is this really needed? Does it compile without these arguments? The simple_output_script thing looks a bit hacky to me and I would like to avoid this unless it requires a considerable amount of work.

Reply:
usually, you may bypass them by specifying `HELP2MAN=:` or 'HELP2MAN=/bin/true`. but it might not be enough for this particular case (I don't remember, it needs checking).

Reply:
@madebr you replaced this in https://github.com/conan-io/conan-center-index/pull/1196/commits/13aa26221a680e23fdb82037bf954a43b1b85868 Do you remember why ?
Without HELPAMN and MAKEINFO overriding the build passes fine on my windows and linux configuration (without help2man and makeinfo executables), and on CCI too. is it really needed ?